common/namespace/mutate.go

```diff
@@ -26,70 +26,93 @@ package namespace
 
 import (
 	namespacepb "go.temporal.io/api/namespace/v1"
+	persistencespb "go.temporal.io/server/api/persistence/v1"
+	"go.temporal.io/server/common"
 	"go.temporal.io/server/common/persistence"
 	"google.golang.org/protobuf/types/known/durationpb"
 )
 
-type mutationFunc func(*persistence.GetNamespaceResponse)
+type mutationFunc func(*Namespace)
```

Comment:
Mutations now apply to the namespace itself, itself of (indirectly) on the persistence response.